pipelines/tasks/StarMkref.wdl

```diff
@@ -0,0 +1,31 @@
+
+task StarMkref {
+  File fasta_file  # genome annotation
+  File annotation_file  # gtf annotation file
+
+  # optionally, use a different version of star
+  String star_docker_image = "ambrosejcarr/star-ubam:v1.1.0"
+  
+  command {
+    mkdir genome
+    STAR \
+      --runMode genomeGenerate \
+      --runThreadN $(nproc) \
+      --genomeDir genome \
+      --genomeFastaFiles ${fasta_file} \
+      --sjdbGTFfile ${annotation_file} \
+
+    echo "tarring genome" && tar -czf genome.tar.gz genome/*
+  }
+  
+  runtime {
+    docker: "${star_docker_image}"
+    cpu: 16
+    memory: "30 GB"
+    disks: "local-disk 100 SSD"
+  }
+  
+  output {
+    File genome = "genome.tar.gz"
```

Comment:
so you are going to tar.gz reference then? I remember the gz file will take extra time compare to tar file.

Reply:
We should test how long these each take. In my experience, while .gz takes more time to decompress, the reduced size makes them transfer faster and the transfer speeds I've been seeing to GCE instances is _slow_.

Reply:
If I remember correctly, the size difference bwt tar.gz and gz of reference bundle is not significant and but without  unzipping process, it will save about 10~15 min on SS2. But SS2 pipeline is  single-sample per run fashion so gz or un-gz is more significant than 10X I guess.

Reply:
I'll explicitly test. If there is a negative impact or mild positive impact for mine I'll remove the gzip since it sounds like it makes things slower for SS2.

Reply:
You can often separate and parallelize the compression to speed things up. Something like `tar cf - genome/* | pigz > genome.tar.gz`

Reply:
Also, is there any ability to do streaming with WDL/cromwell? Ideally you would be able to do something like
```
stream-input ${input_tar} | gunzip | tar xf - -C genome
```
and then the unzipping and the file transfer can happen simultaneously. But I don't think there's anything like `stream-input` in WDL that would work without abusing types.

Reply:
I don't think this exists. I really wish it would, though. It would make tasks so much faster...

Reply:
pigz -d is a bit slower (~10%) than the download speed I'm getting on the instances. I've removed the gzipping. Thanks for the comment!